src/net/if_.rs

```diff
@@ -0,0 +1,48 @@
+//! Network interface name resolution.
+//!
+//! Uses Linux and/or POSIX functions to resolve interface names like "eth0"
+//! or "socan1" into device numbers.
+
+use libc::{c_char, c_uint};
+use std::ffi::{CString, NulError};
+use std::io;
+
+/// Error that can occur during interface name resolution.
+#[derive(Debug)]
+pub enum NameToIndexError {
+    /// Failed to allocate a C-style string to for the syscall
+    NulError,
+    IOError(io::Error),
+}
+
+impl From<NulError> for NameToIndexError {
+    fn from(_: NulError) -> NameToIndexError {
+        NameToIndexError::NulError
+    }
+}
+
+impl From<io::Error> for NameToIndexError {
+    fn from(e: io::Error) -> NameToIndexError {
+        NameToIndexError::IOError(e)
+    }
+}
+
+extern {
+    fn if_nametoindex(ifname: *const c_char) -> c_uint;
+}
+
+/// Resolve an interface into a interface number.
+pub fn name_to_index(name: &str) -> Result<c_uint, NameToIndexError> {
```

Comment:
> That currently seems a little messy and missing an implementation for &str? Interface names are not Paths and I don't want to require passing in CStrings, because at that point it stops being a Rust-Interface, I think.

This is a good consideration. I've copied your note over to to #221 where we're discussing string handling in nix.